plugin/builtingen/ImageTagTransformer.go

```diff
@@ -0,0 +1,40 @@
+// Code generated by pluginator on ImageTagTransformer; DO NOT EDIT.
+package builtingen
+
+import (
+	"sigs.k8s.io/kustomize/pkg/ifc"
+	"sigs.k8s.io/kustomize/pkg/image"
+	"sigs.k8s.io/kustomize/pkg/resmap"
+	"sigs.k8s.io/kustomize/pkg/transformers"
+	"sigs.k8s.io/kustomize/pkg/transformers/config"
+	"sigs.k8s.io/yaml"
+)
+
+// Find matching image declarations and replace
+// the name, tag and/or digest.
+type ImageTagTransformerPlugin struct {
+	ImageTag   image.Image        `json:"imageTag,omitempty" yaml:"imageTag,omitempty"`
+	FieldSpecs []config.FieldSpec `json:"fieldSpecs,omitempty" yaml:"fieldSpecs,omitempty"`
+}
+
+func NewImageTagTransformerPlugin() *ImageTagTransformerPlugin {
+  return &ImageTagTransformerPlugin{}
+}
+
+func (p *ImageTagTransformerPlugin) Config(
+	ldr ifc.Loader, rf *resmap.Factory, c []byte) (err error) {
+	p.ImageTag = image.Image{}
+	p.FieldSpecs = nil
+	return yaml.Unmarshal(c, p)
+}
+
+func (p *ImageTagTransformerPlugin) Transform(m resmap.ResMap) error {
+	argsList := make([]image.Image, 1)
+    argsList[0] = p.ImageTag
+    t, err := transformers.NewImageTransformer(argsList, p.FieldSpecs)
```

Comment:
So we will keep all existing transformers as they are right now. The transformer plugin uses those existing transformers. Is this to demonstrate how to use existing transformers?

Reply:
There's a TODO to move the code into the plugins.

I think this would inform which kustomize packages should remain public, and which should move into internal.  Right now, we have  easy to understand means to make this choice, so we don't make a choice, and leave everything public. A bad, hard to support API.